marketplace/core/types/channels/whatsapp_cloud/views.py

```diff
@@ -49,7 +49,26 @@ def get_queryset(self):
         return super().get_queryset().filter(code=self.type_class.code)
 
     def destroy(self, request, *args, **kwargs) -> Response:
-        return Response("This channel cannot be deleted", status=status.HTTP_403_FORBIDDEN)
+        serializer = WhatsAppCloudDeleteSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        headers = {"Authorization": f"Bearer {settings.WHATSAPP_SYSTEM_USER_ACCESS_TOKEN}"}
+        base_url = settings.WHATSAPP_API_URL
+        phone_number_id = serializer.validated_data.get("phone_number_id")
+
+        url = f"{base_url}/{phone_number_id}/deregister"
+    
+        response = requests.post(url, headers=headers)
+
+        if response.status_code != status.HTTP_200_OK:
+            raise ValidationError(response.json())
+
+        task = celery_app.send_task(
+            name="release_channel", args=[serializer.validated_data.get("channel_uuid"), serializer.validated_data.get("user_email")]
+        )
+        task.wait()
+
```

Comment:
You need to delete the App.
With `self.get_object()` you can get the App instance